code/game/turfs/open.dm

```diff
@@ -347,6 +347,9 @@
 
 		else if (isliving(thing))
 			var/mob/living/L = thing
+			if(L.status_flags & GODMODE) //Invulnerable mobs shouldn't care/take damage
```

Comment:
instead of this, add this godmode check to adjust_fire_stacks() and IgniteMob().  take_overall_damage() already ignores damage if the mob is godmode